### DISCUSSION TITLE
Automerge using regexManagers (customManagers)

### DISCUSSION THREAD

**Discussion by adamxkabbara** [renovatebot/renovate#29850] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

None

### Please tell us more about your question or problem

I am using regexManagers and wanted to know if there is a way to use the automerge feature renovate provides. I know there is a way for packateRules but need it for regexManagers.

Here is a rough template I am using.
{
 "regexMangers": [
   ...
 ]
}

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Yes you can. You enable it the usual way using packageRules

**Reply by adamxkabbara** [reply]
@rarkins Sorry, I am a novice when it comes to renovate. How would packageRules know which regexManager I am referring to? Is there a property like an Id to assign to the regexManager that I can reference in packageRules?

**Reply by rarkins** [reply, marked as answer]
You would need to use a combination of matchManagers, matchDatasources, and matchPackageNames (or similar)

**Reply by RahulGautamSingh** [reply]
Example:
```json
{
  "packageRules": [
    {
      "matchDatasources": ["npm"],
      "matchPackageNames": ["lodash"],  // or
      "matchManagers": ["custom.regex"] // or
      "automerge": true
    }
  ]
}
```
